lightning/src/ln/peer_handler.rs

```diff
@@ -1272,4 +1284,124 @@ mod tests {
+				let chan_upd_2 = get_dummy_channel_update(j);
+				let chan_ann = get_dummy_channel_announcement(j);
+
+				chan_anns.push((chan_ann, chan_upd_1, chan_upd_2));
+			}
+
+			chan_anns
+		}
+
+		fn get_next_node_announcements(&self, _starting_point: Option<&PublicKey>, _batch_amount: u8) -> Vec<msgs::NodeAnnouncement> {
+			Vec::new()
+		}
+
+		fn should_request_full_sync(&self, _node_id: &PublicKey) -> bool {
+			true
+		}
+	}
+
+	fn get_dummy_channel_announcement(short_chan_id: u64) -> msgs::ChannelAnnouncement {
+		let secp_ctx = Secp256k1::new();
+		let network = Network::Testnet;
+		let node_1_privkey = SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_2_privkey = SecretKey::from_slice(&[41; 32]).unwrap();
+		let node_1_btckey = SecretKey::from_slice(&[40; 32]).unwrap();
+		let node_2_btckey = SecretKey::from_slice(&[39; 32]).unwrap();
+		let unsigned_ann = msgs::UnsignedChannelAnnouncement {
+			features: ChannelFeatures::supported(),
+			chain_hash: genesis_block(network).header.bitcoin_hash(),
+			short_channel_id: short_chan_id,
+			node_id_1: PublicKey::from_secret_key(&secp_ctx, &node_1_privkey),
+			node_id_2: PublicKey::from_secret_key(&secp_ctx, &node_2_privkey),
+			bitcoin_key_1: PublicKey::from_secret_key(&secp_ctx, &node_1_btckey),
+			bitcoin_key_2: PublicKey::from_secret_key(&secp_ctx, &node_2_btckey),
+			excess_data: Vec::new(),
+		};
+
+		let msghash = Message::from_slice(&Sha256dHash::hash(&unsigned_ann.encode()[..])[..]).unwrap();
+		let node_1_sig = secp_ctx.sign(&msghash, &node_1_privkey);
+		let node_2_sig = secp_ctx.sign(&msghash, &node_2_privkey);
+		let bitcoin_1_sig = secp_ctx.sign(&msghash, &node_1_btckey);
+		let bitcoin_2_sig = secp_ctx.sign(&msghash, &node_2_btckey);
+
+		msgs::ChannelAnnouncement {
+			node_signature_1: node_1_sig,
+			node_signature_2: node_2_sig,
+			bitcoin_signature_1: bitcoin_1_sig,
+			bitcoin_signature_2: bitcoin_2_sig,
+			contents: unsigned_ann,
+		}
+	}
+
+	fn get_dummy_channel_update(short_chan_id: u64) -> msgs::ChannelUpdate {
+		use secp256k1::ffi::Signature as FFISignature;
+		let network = Network::Testnet;
+		msgs::ChannelUpdate {
+			signature: Signature::from(FFISignature::new()),
+			contents: msgs::UnsignedChannelUpdate {
+				chain_hash: genesis_block(network).header.bitcoin_hash(),
+				short_channel_id: short_chan_id,
+				timestamp: 0,
+				flags: 0,
+				cltv_expiry_delta: 0,
+				htlc_minimum_msat: 0,
+				fee_base_msat: 0,
+				fee_proportional_millionths: 0,
+				excess_data: vec![],
+			}
+		}
+	}
+
+	#[test]
+	fn test_do_attempt_write_data() {
+		// Create 2 peers with custom TestRoutingMessageHandlers and connect them.
+		let chan_handlers = create_chan_handlers(2);
+		let mut routing_handlers: Vec<Arc<msgs::RoutingMessageHandler>> = Vec::new();
+		for i in 0..2 {
+			let routing_handler_option1 = TestRoutingMessageHandler::new();
+			let routing_handler_option2 = TestRoutingMessageHandler::new();
+			if i == 0 {
```

Comment:
Hmm? So for each loop iteration we create two msgs::RoutingMessageHandlers, then on iteration one we use one and throw away the other and on iteration two we use the second and throw away the first? I assume there was something different in a previous iteration?

Reply:
Yeah, 'twas outdated